SPECS/glusterfs/glusterfs.spec

```diff
@@ -151,346 +200,411 @@ BuildRequires:  libattr-devel
+Requires:         %{name}%{?_isa} = %{version}-%{release}
+Requires:         %{name}-client-xlators%{?_isa} = %{version}-%{release}
+Requires:         nfs-utils
 
 %description gnfs
 GlusterFS is a distributed file-system capable of scaling to several
-petabytes. It aggregates various storage bricks over Infiniband RDMA
-or TCP/IP interconnect into one large parallel network file
-system. GlusterFS is one of the most sophisticated file systems in
-terms of features and extensibility.  It borrows a powerful concept
-called Translators from GNU Hurd kernel. Much of the code in GlusterFS
-is in user space and easily manageable.
+petabytes. It aggregates various storage bricks over TCP/IP interconnect
+into one large parallel network filesystem. GlusterFS is one of the
+most sophisticated file systems in terms of features and extensibility.
+It borrows a powerful concept called Translators from GNU Hurd kernel.
+Much of the code in GlusterFS is in user space and easily manageable.
 
 This package provides the glusterfs legacy gNFS server xlator
 %endif
 
-%package libs
-Summary:        GlusterFS common libraries
-License:        GPLv2 OR LGPLv3+
+%package -n libglusterfs0
+Summary:          GlusterFS libglusterfs library
+Requires:         libgfrpc0%{?_isa} = %{version}-%{release}
+Requires:         libgfxdr0%{?_isa} = %{version}-%{release}
+Obsoletes:        %{name}-libs <= %{version}-%{release}
+Provides:         %{name}-libs = %{version}-%{release}
+
+%description -n libglusterfs0
+GlusterFS is a distributed file-system capable of scaling to several
+petabytes. It aggregates various storage bricks over TCP/IP interconnect
+into one large parallel network filesystem. GlusterFS is one of the
+most sophisticated file systems in terms of features and extensibility.
+It borrows a powerful concept called Translators from GNU Hurd kernel.
+Much of the code in GlusterFS is in user space and easily manageable.
+
+This package provides the base libglusterfs library
+
+%package -n libglusterfs-devel
+Summary:          GlusterFS libglusterfs library
+Requires:         libgfrpc-devel%{?_isa} = %{version}-%{release}
+Requires:         libgfxdr-devel%{?_isa} = %{version}-%{release}
+Obsoletes:        %{name}-devel <= %{version}-%{release}
+Provides:         %{name}-devel = %{version}-%{release}
+
+%description -n libglusterfs-devel
+GlusterFS is a distributed file-system capable of scaling to several
+petabytes. It aggregates various storage bricks over TCP/IP interconnect
+into one large parallel network filesystem. GlusterFS is one of the
+most sophisticated file systems in terms of features and extensibility.
+It borrows a powerful concept called Translators from GNU Hurd kernel.
+Much of the code in GlusterFS is in user space and easily manageable.
+
+This package provides libglusterfs.so and the gluster C header files.
+
+%package -n libgfapi0
+Summary:          GlusterFS api library
+Requires:         libglusterfs0%{?_isa} = %{version}-%{release}
```

Comment:
There's quite a bunch of package additions and reshuffling happening in this spec and I can't get through all of the dependencies they have. Can you tell me, if in case of new dependencies (assuming there are any), they will be satisfied by our packages? I'm a bit on the fence when it comes to introducing new packages (I think there are some although most only look like a re-name of existing ones) but if we're going to do so, we must make sure we have their run-time dependencies.

Reply:
i'm not sure how to do what you're asking or what it refers to.  can you provide more info?

Reply:
from a quick scan of the Requires

```
$ cat ./SPECS/glusterfs/glusterfs.spec | grep -v '^\-' | grep -v BuildRequir | grep Requires | cut -c19- | awk '{print $1}' | sort --unique
attr
dbus
firewalld-filesystem
gperftools-libs%{?_isa}
libacl-devel
%{_libdir}/ocf/resource.d
libgfapi0%{?_isa}
libgfchangelog0%{?_isa}
libgfrpc0%{?_isa}
libgfrpc-devel%{?_isa}
libgfxdr0%{?_isa}
libgfxdr-devel%{?_isa}
libglusterfs0%{?_isa}
libglusterfs-devel%{?_isa}
lvm2
%{name}-client-xlators%{?_isa}
%{name}-cli%{?_isa}
%{name}-fuse%{?_isa}
%{name}%{?_isa}
%{name}-server
%{name}-server%{?_isa}
nfs-ganesha-gluster
nfs-ganesha-selinux
nfs-utils
pcs
portmap
psmisc
python%{_pythonver}
python%{_pythonver}-gluster
python%{_pythonver}-prettytable
python%{_pythonver}-requests
pyxattr
resource-agents
rsync
shadow-utils
tar
util-linux
valgrind
```

to me, these seem to be the only ones that are unfulfilled:

nfs-ganesha-gluster [package ganesha]
nfs-ganesha-selinux [package ganesha]
pcs [package ganesha]
pyxattr (in EXTENDED) [package server]

Reply:
server is disabled.  maybe it should be deleted from the spec rather than disabled?

i'm not sure what it is, but maybe we disable/delete the ganesha package too?

Reply:
removing ganesha package